agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/feature/config/FeatureConfig.kt

```diff
@@ -12,29 +12,25 @@ import ai.koog.agents.core.feature.message.FeatureMessageProcessor
  */
 public abstract class FeatureConfig {
 
-    private val _messageProcessors = mutableListOf<FeatureMessageProcessor>()
-
-    private var _eventFilter: (AgentLifecycleEventContext) -> Boolean = { true }
-
-    /**
-     * Provides a read-only list of `FeatureMessageProcessor` instances registered with the feature configuration.
-     */
-    public val messageProcessors: List<FeatureMessageProcessor>
-        get() = _messageProcessors.toList()
-
     /**
-     * A filter for events to be processed by a feature.
+     * A list of [FeatureMessageProcessor] instances registered with the feature configuration.
      */
-    public val eventFilter: (AgentLifecycleEventContext) -> Boolean
-        get() = _eventFilter
+    public var messageProcessors: List<FeatureMessageProcessor> = emptyList()
+        private set
 
     /**
      * Adds a message processor to the configuration.
      */
     public fun addMessageProcessor(processor: FeatureMessageProcessor) {
-        _messageProcessors.add(processor)
+        messageProcessors = messageProcessors + processor
     }
 
+    /**
+     * A filter for events to be processed by a feature.
+     */
+    public var eventFilter: (AgentLifecycleEventContext) -> Boolean = { true }
```

Comment:
As we discussed, this explicit functions were added here for readability and follow the same pattern through feature configs. If we agree to change it, let's deprecate the old methods in that case.